### ISSUE TITLE
Cleanup the .zip artifact?

### ISSUE THREAD

**Issue by p-bakker** [mozilla/rhino#1119]
This is what is inside the Rhino 1.7.13 folder in the .zip artifact for  v1.7.13:
![image](https://user-images.githubusercontent.com/4981149/145367481-38e57b64-455d-497c-a821-d127eb0ae046.png)

I think of that, the following should not be included:
- CODE_OF_CONDUCT.md
- README.md
- RELEASE-NOTES.md
- build.gradle
- docs/
- gradle/
- gradle.properties
- gradlew*
- release-steps.txt
- results.txt

And shouldn't the /lib folder not also include the `rhino-runtime-X.X.X.jar`, the `rhino-engine-X.X.X.jar` and the toolsrc/xmlimplsrc folders?

**Comment by carlosame** [comment]
Until now, I haven't noticed that the Javadocs are part of the source Git repository, at `docs/javadoc/`. Shouldn't they be removed, and instead produce a javadoc jar on `./gradlew build`? (the javadocs are already being generated at `buildGradle/docs/javadoc/`)

**Comment by p-bakker** [comment]
Yeah, not very clear that we're hosting the javadoc, but that's being worked on

Why wouldn't we host them and instead produce a jar?

**Comment by p-bakker** [comment]
That was recently switched: the GitHub pages are now based on the docs folder in master

**Comment by gbrail** [comment]
I personally find having a "docs" folder in master a much much easier way
to manage the (rather sparse) pages at mozilla.github.io/rhino. If there's
a better way to ensure that Javadocs are hosted somewhere, then I'm happy
to have some show us how to switch to that, but for now it's helpful IMO to
have Javadoc online somewhere and right now that's the easiest place to put
it!

The "zip" artifact is something that I don't know how people use -- it's
always been like that, and I've been told before that changing it breaks
various workflows, but in these days of people building with Gradle and
Maven I wonder what it's really for. Does anyone have an anecdote?

**Comment by tonygermano** [comment]
> PErsonally also don't have a problem hosting the JavaDoc there, only wondering if we should not only host the latest version, but also (some) historic versions

+1 for hosting historic versions

**Comment by tonygermano** [comment]
Is the zip artifact the same thing as if you were to go to the main project page and click the `Code` button (the same one that gives you the clone options, and select the "Download ZIP" option instead?

**Comment by tonygermano** [comment]
I honestly can say that I've never looked at that artifact before. I guess I'd say don't mess with it if we don't know who is using it and why. Or just stop publishing it and see who complains.

I don't think it hurts to leave the gradle stuff in there since it includes the source files, too.

The xmlimpl and tools packages appear to be moved under the main src dir in that zip file.

**Comment by p-bakker** [comment]
Wow... https://javadoc.io/doc/org.mozilla/rhino/latest/index.html

Lets remove the Javadoc from within the /docs folder...

**Comment by gbrail** [comment]
That's awesome -- we should just change the README link. I might be
running out of time for Rhino for a few days but I'll try to get to it, and
of course I'm accepting PRs! ;-)

**Comment by p-bakker** [comment]
Back to the .zip file: I'd be in favor of removing it, as its a confusing file and conveys the wrong message, as it only includes the full jar, which is not what most people should be using.

Secondly, having the source in there doesn't make sense either: if someone wants the source, they should clone the repo or download the zip file that GitHub provides: the source in the .zip artifact/assets is mangled by inlining the toolsrc and xmlimplsrc, so doesn't represent the source structure as we have it in the repo

This is with the assumption that most people embed Rhino and not use it for its Shell.

What embedders should be using is the rhino-runtime jar, potentially combined with the rhino-engine jar if they want to use Rhino through the JSR-223 SCript Engine interface.

Right now (based on what used to be in the developer.mozilla.org/rhino/... documentation), I'm referring to the .zip file for every release:, see https://p-bakker.github.io/rhino/docs/releases/index/#release-overview. But I think I should stop doing that.

> Or just stop publishing it and see who complains.

I'd be in favor :-)

If you look at what historically was included, it has changes significaltly over time, form just src & example (1.4R3), to all individual *src folders + examples + javadoc (1.7.R4), to one merged src folder + docs (containing only the JavaDoc (1.7.13) etc.

So I wonder who would depend exactly on what...

**Comment by carlosame** [comment]
> What embedders should be using is the rhino-runtime jar

That jar is unusable with modular JDKs because the packages that it contains are also present in the main rhino jar (which is likely to be in the same modulepath, due to it being somebody else's transitive dependency). The solution is #1092.

**Comment by p-bakker** [comment]
See https://github.com/mozilla/rhino/issues/1120#issuecomment-990909170

Our repo contains a(n outdated) manpage for the Shell, but due to a misconfiguration in build.gradle it's not being shipped in any of our artifacts.

As per @carlosame [question](https://github.com/mozilla/rhino/issues/1120#issuecomment-990925649): should we ship this manpage (and if so how) or should we toss it?

**Comment by p-bakker** [comment]
> The solution is #1092.

I think once 1.7.14 has been released, #1092 should be part of the [discussions](#1066)

**Comment by carlosame** [comment]
Added an updated manpage to #1124. The "see also" URLs are from archive.org, could not find anything else.

**Comment by p-bakker** [comment]
Noticed this line in our Readme.md:

> There is also a "rhino" package for many Linux distributions as well as Homebrew for the Mac.

Which got me wondering and after some searching found I found, amongst others, this: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/rhino.rb

Looks like stuff like this uses the .zip artifact.
And maybe the .man/rhino.1 page would also get used in a similar way on Linux packages. 

Anyone experience with that sort of stuff? Any idea how such OS specific packaging systems get updated? Are we doing that somehow?

**Comment by p-bakker** [comment]
On on Fedora, they seem to build their own jars, based on the POM's from Maven and downloading the entire git repository: https://src.fedoraproject.org/rpms/rhino/blob/rawhide/f/rhino.spec

They use the man/rhino.1 page

**Comment by carlosame** [comment]
> Looks like stuff like this uses the .zip artifact. And maybe the .man/rhino.1 page would also get used in a similar way on Linux packages.

Yes I think so.

> Anyone experience with that sort of stuff? Any idea how such OS specific packaging systems get updated? Are we doing that somehow?

> On on Fedora, they seem to build their own jars, based on the POM's from Maven and downloading the entire git repository

Rhino's packages for Fedora are up-to-date (they are at 1.7.13), and perhaps if the zip/tar distribution shipped with the man page they would not need to obtain the entire repository.

Other distributions are stuck at earlier versions. Debian (and Ubuntu which is based on it) and almost everyone else are still at 1.7.7:

https://packages.debian.org/sid/rhino

The following URL gives the Rhino versions shipped by a number of distributions:

https://pkgs.org/download/rhino
